visualization/app/codeCharta/codeCharta.service.ts

```diff
@@ -1,39 +1,63 @@
-import { FileValidator, CCValidationResult } from "./util/fileValidator"
+import { validate } from "./util/fileValidator"
 import { AttributeTypes, CCFile, NameDataPair, BlacklistType, BlacklistItem } from "./codeCharta.model"
 import _ from "lodash"
 import { NodeDecorator } from "./util/nodeDecorator"
+import { ExportBlacklistType, ExportCCFile, OldAttributeTypes } from "./codeCharta.api.model"
 import { StoreService } from "./state/store.service"
 import { addFile, setSingle } from "./state/store/files/files.actions"
 import { getCCFiles } from "./model/files/files.helper"
+import { DialogService } from "./ui/dialog/dialog.service"
+import { setIsLoadingFile } from "./state/store/appSettings/isLoadingFile/isLoadingFile.actions"
+import { setState } from "./state/store/state.actions"
+import { ScenarioHelper } from "./util/scenarioHelper"
+import { MetricService } from "./state/metric.service"
 
 export class CodeChartaService {
 	public static ROOT_NAME = "root"
 	public static ROOT_PATH = "/" + CodeChartaService.ROOT_NAME
 	public static readonly CC_FILE_EXTENSION = ".cc.json"
 
-	constructor(private storeService: StoreService) {}
-
-	public loadFiles(nameDataPairs: NameDataPair[]): Promise<void> {
-		return new Promise((resolve, reject) => {
-			nameDataPairs.forEach((nameDataPair: NameDataPair) => {
-				const validationResult: CCValidationResult = FileValidator.validate(nameDataPair.content)
-				if (validationResult.error.length === 0) {
-					const ccFile = this.getCCFile(nameDataPair.fileName, nameDataPair.content)
-					NodeDecorator.preDecorateFile(ccFile)
-					this.storeService.dispatch(addFile(ccFile))
-					if (validationResult.warning.length !== 0) {
-						reject(validationResult)
-					}
-				} else {
-					reject(validationResult)
+	constructor(private storeService: StoreService, private dialogService: DialogService, private metricService: MetricService) {}
+
+	public loadFiles(nameDataPairs: NameDataPair[]) {
+		let printErrors = true
+
+		nameDataPairs.forEach((nameDataPair: NameDataPair) => {
+			if (!printErrors) {
+				return
+			}
+
+			try {
+				validate(nameDataPair.content)
+				this.addFile(nameDataPair.fileName, nameDataPair.content)
+			} catch (e) {
+				if (!_.isEmpty(e.error)) {
+					this.dialogService.showValidationErrorDialog(e)
+					printErrors = false
```

Comment:
Should this only stop in case of errors, not warnings?

We miss to remove already added files in case the first file is valid but the second one is not. This might cause issues when trying to load files again after having validation errors.

Reply:
> Should this only stop in case of errors, not warnings?

Yes. Warnings are totally fine. Warnings happen when a file is outdated. But since we're backward compatible, we can read the file.

> We miss to remove already added files in case the first file is valid but the second one is not. This might cause issues when trying to load files again after having validation errors.

That is the case in this PR, but is adressed in #1082 . We wait until all files are validated before setting them. Let's fix this issue in #1082 then